### PR TITLE
Experimentation pay gate

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

Implements a frontend pay gate for Experimentation. If you don't have the feature available:

1. You will see this in the Experiments page (this was based on @clarkus's design, abstracted the existing Group Analytics upsell page to a reusable component).

2. If you visit `/experiments/{id}` or `/experiments/new` you'll be taken to the main experiments page where you will see the upsell.

3. No API requests will be made if you don't have the feature available.

In addition,
- Prepared this launch banner for experimentation (and this now removes the Group Analytics banner, FYI @macobo) but set it to hidden for now. Let's launch as beta and then do the announcement.

## How did you test this code?
- Test with

### CODE REVIEW

File: frontend/src/lib/introductions/NewFeatureBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { LinkButton } from 'lib/components/LinkButton'
+import { Link } from 'lib/components/Link'
+import { userLogic } from 'scenes/userLogic'
+
+export function NewFeatureBanner(): JSX.Element | null {
+    const { upgradeLink } = useValues(userLogic)
+
+    return (
+        <div>
+            <strong>🧪 Introducing Experimentation!</strong> Test changes to your product and how they impact your
```
Comment (paolodamico) [inline frontend/src/lib/introductions/NewFeatureBanner.tsx:12]: @joethreepwood for 👀 on copy

File: frontend/src/scenes/experiments/Experiments.tsx [hunk 2]
```diff
@@ -79,29 +83,56 @@ export function Experiments(): JSX.Element {
...
+                <PayGatePage
+                    featureKey="experimentation"
+                    header={
+                        <>
+                            Introducing <span className="highlight">Experimentation</span>!
+                        </>
+                    }
+                    caption={
+                        <>
+                            Test changes to your product and how they impact your users. A/B test new features and
```
Comment (paolodamico) [inline frontend/src/scenes/experiments/Experiments.tsx:129]: @joethreepwood for 👀 on copy

**Comment by paolodamico** [comment]
@neilkakkar what's the best way to mock in `experimentLogic.test.ts` that you have the premium feature enabled?

File: frontend/src/lib/components/PayGatePage/PayGatePage.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                )}
+                {docsLink && (
+                    <LinkButton
+                        type={hideUpgradeButton ? 'primary' : undefined}
+                        to={`${docsLink}?utm_medium=in-product&utm_campaign=${featureKey}-upgrade-learn-more`}
+                        target="_blank"
+                        data-attr={`${featureKey}-learn-more`}
+                        className="LemonLinkButton"
+                    >
+                        Check the docs. Learn how to use <IconOpenInNew style={{ marginLeft: 8 }} />
```
Comment (clarkus) [inline frontend/src/lib/components/PayGatePage/PayGatePage.tsx:49]: This button copy isn't really aligned with our other upsells. I think it should follow what you have for the banner - something like "Learn more about experimentation". Thoughts?

**Comment by neilkakkar** [comment]
Something like this! https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/funnels/funnelLogic.test.ts#L224

(although a subsequent PR that's currently in review will remove most of the tests using things - as calcs have moved to the backend)

File: frontend/src/scenes/experiments/Experiments.tsx (hunk 2 above)
Comment (joethreepwood) [reply frontend/src/scenes/experiments/Experiments.tsx:129]: LGTM

File: frontend/src/lib/introductions/NewFeatureBanner.tsx (hunk 1 above)
Comment (joethreepwood) [reply frontend/src/lib/introductions/NewFeatureBanner.tsx:12]: LGTM

File: frontend/src/scenes/experiments/Experiments.tsx (hunk 2 above)
Comment (joethreepwood) [reply frontend/src/scenes/experiments/Experiments.tsx:129]: There's some repetition here. I also think 'improve your metrics' as a goal is a bit cynical compared to 'build better products' 

```suggestion
                            A/B test changes and new features to see how they impact your users and improve your product!
```

Or, the slightly more active

```suggestion
                            Improve your product by A/B testing new features to discover what works best for your users.
```

File: frontend/src/scenes/experiments/Experiments.tsx [hunk 4]
```diff
@@ -79,29 +83,56 @@ export function Experiments(): JSX.Element {
...
+                            changes to discover what improves your metrics best.
```
Comment (joethreepwood) [inline frontend/src/scenes/experiments/Experiments.tsx:130]: Would have to adapt/remove this line to reflect suggestions above.

File: frontend/src/lib/components/PayGatePage/PayGatePage.tsx (hunk 3 above)
Comment (paolodamico) [reply frontend/src/lib/components/PayGatePage/PayGatePage.tsx:49]: This is actually now an abstracted component so Groups would've been updated too, that's the reason for the generic copy, will update

**Review by neilkakkar — APPROVED** [review]
Looks good!

**Pull request merged**